clients/src/test/java/org/apache/kafka/common/requests/RequestTestUtils.java

```diff
@@ -208,10 +209,10 @@ public static MetadataResponse metadataUpdateWith(final String clusterId,
             for (int i = 0; i < numPartitions; i++) {
                 TopicPartition tp = new TopicPartition(topic, i);
                 Node leader = nodes.get(i % nodes.size());
-                List<Integer> replicaIds = Collections.singletonList(leader.id());
+                List<Integer> replicaIds = nodes.stream().map(Node::id).collect(Collectors.toList());
```

Comment:
From @hachikuji: Leaving the replication factor as implicit seems less than ideal. Perhaps we could make it an explicit argument?